test/e2e/e2e_suite_test.go

```diff
@@ -136,6 +142,9 @@ func TestE2E(t *testing.T) {
 
 // Create a kind cluster that shared across all the tests
 var _ = SynchronizedBeforeSuite(func() []byte {
+	// This line prevents controller-runtime from complaining about log.SetLogger never being called
+	ctrl.SetLogger(klog.Background())
```

Comment:
Does controller-runtime become direct go mod dependency because of this?

Reply:
It only impacts the e2e test module, not for the main `go.mod`. 
The test module is leveraging cluster-api and  `controller-runtime`  anyway, we should be good.